### PR TITLE
EIP-1682: Storage Rent

### PR DESCRIPTION
**Pull request by holiman**
This PR includes an EIp about storage rent. I'm not sure who else should be added / wants to be added to the author-field, there were many people involved. 

cc @fjl @AlexeyAkhunov @cdetrio @axic @karalabe

### CODE REVIEW

File: EIPS/eip-draft_storagerent.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    else:
+        fee -= poststate[addr].rentbalance
+        poststate[addr].rentbalance = 0
+        poststate[addr].balance -= min(poststate[addr].balance, fee)
+    poststate[addr].rentblock = currentblock
+```
+
+### New EVM Opcodes
+
+#### `PAYRENT <balance> <addr>`
```
Comment (veox) [inline EIPS/eip-draft_storagerent.md]: You likely mean `<amount>`, not `<balance>`.

**Comment by kevinsimper** [comment]
I read the eip and I don't get from reading if the ether should be burnt or transfered. Has that been thought about?

**Comment by holiman** [comment]
From the specification section: 

> The cost of storing an account over time is called `rent`. The amount of `rent` due depends
on the size of the account. The `ether` that is paid for `rent` is destroyed.

**Comment by nicksavers** [comment]
@holiman Is this still relevant in light of https://github.com/ethereum/EIPs/issues/1418 and @AlexeyAkhunov 's proposal?

File: EIPS/eip-1682.md [hunk 2]
```diff
@@ -0,0 +1,208 @@
+---
+eip: 1682
+title: Storage Rent
+author: Felix J Lange (@fjl), Martin Holst Swende (@holiman)
+discussions-to: <URL>
```
Comment (axic) [inline EIPS/eip-1682.md]: Needs a discussion URL.

**Comment by axic** [comment]
@holiman did you wanted to propose this for Istanbul?

**Comment by holiman** [comment]
@axic No, I don't think so. This represents what we thought up back at devcon. It was one of the earlier versions of storage rent, which @AlexeyAkhunov has since brought a long way ahead.

**Comment by axic** [comment]
Should this be merged with a `Rejected` status then (or `Abandoned` as per #2051)?

**Comment by holiman** [comment]
Ah, well, maybe Abandoned then?

**Comment by axic** [comment]
@holiman #2051 was merged, you can mark this Abandoned if you want it to be merged officially.  If not, perhaps we should just close this PR.

File: EIPS/eip-1682.md (hunk 2 above)
Comment (axic) [reply EIPS/eip-1682.md]: Apparently this is the correct one: https://ethereum-magicians.org/t/storage-rent-eip/2357

File: EIPS/eip-1682.md [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+status: Draft
+type: Standards Track
+category: Core
+created: 2018-11-10
+---
+
+## Simple Summary
+
+This EIP describes a scheme to charge for data in state, and 'archive' data which is no longer being paid for. It also describes how resurrection of 'archived' data happens.   
+
```
Comment (axic) [inline EIPS/eip-1682.md]: Please also add an abstract.

**Comment by axic** [comment]
@holiman do you want to "finish" this as discussed above?

**Comment by holiman** [comment]
Yes, sorry, I will

**Comment by axic** [comment]
@holiman made the changes myself, can you confirm your agreement to them?

**Comment by holiman** [comment]
I approve!

**Comment by axic** [comment]
Thanks! One less pending PR :wink:

**Pull request merged**